roles/keycloak/vars/main.yml

```diff
@@ -0,0 +1,78 @@
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+_keycloak_helm_values:
+  # Note(okozachenko1203): Mysql vendor is not supported by bitnami helm chart. As a workaround,
+  #                        we have to define jdbc connection string explicitly along side
+  #                        `externalDatabase` helm values.
+  extraEnvVars:
+    - name: KC_TRANSACTION_XA_ENABLED
+      value: "false"
+    - name: KC_DB
+      value: mysql
+    - name: KC_DB_URL
+      value: "jdbc:mysql://{{ openstack_helm_endpoints.oslo_db.hosts.default }}.openstack:3306/{{ keycloak_database_name }}"
+  auth:
+    adminPassword: "{{ keycloak_admin_password }}"
+    adminUser: "{{ keycloak_admin_username }}"
+  externalDatabase:
+    host: "{{ openstack_helm_endpoints.oslo_db.hosts.default }}.openstack"
+    port: 3306
+    database: "{{ keycloak_database_name }}"
+    existingSecret: "keycloak-db-credentials"
+    existingSecretUserKey: username
+    existingSecretPasswordKey: password
+  image:
+    registry: "{{ atmosphere_images['keycloak'] | vexxhost.kubernetes.docker_image('domain') }}"
+    repository: "{{ atmosphere_images['keycloak'] | vexxhost.kubernetes.docker_image('path') }}"
+    tag: "{{ atmosphere_images['keycloak'] | vexxhost.kubernetes.docker_image('tag') }}"
+  ingress:
+    enabled: true
+    ingressClassName: "{{ keycloak_ingress_class_name }}"
+    annotations:
+      cert-manager.io/cluster-issuer: atmosphere
+    hostname: "{{ keycloak_host }}"
+    tls: true
+  postgresql:
+    enabled: false
+  production: true
+  proxy: edge
+  # Note(okozachenko1203): It takes a bit long time for db migration with mysql vendor.
+  readinessProbe:
+    initialDelaySeconds: 60
+    periodSeconds: 10
+    timeoutSeconds: 1
+    failureThreshold: 6
+    successThreshold: 1
+
+_keycloak_realms:
+  - id: Atmosphere
+    clients:
+      - id: "{{ atmosphere_grafana_oauth_client_id | default('grafana') }}"
+        secret: "{{ atmosphere_grafana_oauth_client_secret }}"
+        admin_url: "https://{{ kube_prometheus_stack_grafana_host }}"
+        base_url: "https://{{ kube_prometheus_stack_grafana_host }}"
+        default_client_scopes:
+          - email
+          - offline_access
+          - profile
+          - roles
+        redirect_uris:
+          - "https://{{ kube_prometheus_stack_grafana_host }}/*"
+        root_url: "https://{{ kube_prometheus_stack_grafana_host }}"
+        web_origins:
+          - "https://{{ kube_prometheus_stack_grafana_host }}"
+        standard_flow_enabled: true
+        direct_access_grants_enabled: true
+        protocol: openid-connect
```

Comment:
we can move this into the seperate roles as discussed

also, i think since the realm goes into the url/etc, it would be best if we lower case it.. so id `atmosphere`.